File: BACKEND/PricingEngineApplication/src/main/java/com/example/PricingEngineApplication/service/ComponentService.java
package com.example.PricingEngineApplication.service;



import java.util.List;

import com.example.PricingEngineApplication.entity.Component;

public interface ComponentService {

    Component addComponent(Component component);

    Component updateComponent(Long id, Component component);

    Component updatePrice(Long id, double newPrice);

    Component getComponentById(Long id);

    List<Component> getAllComponents();

    void deleteComponent(Long id);
}

File: FRONTEND/hero-pricing-engine-ui/src/services/componentService.js
import api from "../api/api";

export const getAllComponents=()=>{

    return api.get("/components");

}

export const getComponent=(id)=>{

    return api.get(`/components/${id}`);

}

export const addComponent=(component)=>{

    return api.post("/components",component);

}

export const updateComponent=(id,component)=>{

    return api.put(`/components/${id}`,component);

}

export const deleteComponent=(id)=>{

    return api.delete(`/components/${id}`);

}

export const updatePrice=(id,price)=>{

    return api.patch(`/components/${id}/price?price=${price}`);

}